### PR TITLE
Added branch status warning to diff report

### LINKED ISSUE #1738: Add check to ensure of the branch is up to date with master
Automated diff should check if branch is up to date with master before allowing the run

### PR DESCRIPTION
**Pull request by ubyndr**
Fixes #1738 

I have added `branch_status` step at the top. It checks if the current branch is behind compared to master branch, and make a comment to warn user. It fails after the comment process.

### CODE REVIEW

File: .github/workflows/diff.yml [hunk 1]
```diff
@@ -1,13 +1,40 @@
 name: 'Create ROBOT diffs on Pull requests'
 
-on: 
+on:
```
Comment (hrshdhgd) [inline .github/workflows/diff.yml:3]: Shouldn't this workflow run after every commit made to a PR? I'm making the assumption that every open branch is associated with a PR. So with that in mind, shouldn't there be
```
on:
  pull_request:
    branches: [ main ]
```

Comment (matentzn) [reply .github/workflows/diff.yml:3]: Ah no, this action fires when a specific command is spoken in a comment on the PR, like "kgcl" bot.

**Review by matentzn — APPROVED** [review]

File: .github/workflows/diff.yml [hunk 2]
```diff
@@ -1,13 +1,40 @@
...
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+        uses: NejcZdovc/comment-pr@v1.1.1
+        with:
+          file: "../../warncomment.md"
+      - name: fail
+        if: steps.info.outputs.status >= 1
+        run: |
+          echo "Your branch is $STATUS commit/s behind, please update your branch.   "
+          exit 1
```
Comment (shawntanzk) [inline .github/workflows/diff.yml:35]: does this actually hard fail? I think I tried this once with another strategy and it still continues. But I guess you've tested it :)

Comment (matentzn) [reply .github/workflows/diff.yml:35]: Good point @shawntanzk I have assumed this was tested, but yea better be sure

Comment (ubyndr) [reply .github/workflows/diff.yml:35]: @shawntanzk unless you use `continue-on-error: true` or `exit 0` it interrupts the run.

**Comment by bvarner-ebi** [comment]
Hi, @ubyndr, checking to see if this PR can be merged. 
If there are pending items, could you list them so they can be addressed?

**Comment by ubyndr** [comment]
> Hi, @ubyndr, checking to see if this PR can be merged. If there are pending items, could you list them so they can be addressed?

Hi @bvarner-ebi , it can be merged.

**Pull request merged**